packages/ontario-design-system-component-library/src/components/ontario-task/ontario-task.tsx

```diff
@@ -0,0 +1,221 @@
+	 * Watch for changes in `taskStatus` prop to validate its value.
+	 */
+	@Watch('taskStatus')
+	validateTaskStatus() {
+		const isValidStatus = validateValueAgainstArray(this.taskStatus, TaskStatuses);
+		this.taskStatusState = isValidStatus ? this.taskStatus : this.warnDefaultTaskStatus();
+	}
+
+	/**
+	 * Display a console warning if `taskStatus` is invalid and set to a default value.
+	 */
+	private warnDefaultTaskStatus(): TaskStatus {
+		const message = new ConsoleMessageClass();
+		message
+			.addDesignSystemTag()
+			.addMonospaceText(' taskStatus ')
+			.addRegularText('on')
+			.addMonospaceText(' <ontario-task> ')
+			.addRegularText('was set to an invalid taskStatus; only ')
+			.addMonospaceText(TaskStatuses.join(', '))
+			.addRegularText(' are supported. The default taskStatus ')
+			.addMonospaceText('notStarted')
+			.addRegularText(' is assumed.')
+			.printMessage();
+		return 'notStarted';
+	}
+
+	/**
+	 * Watch for changes in `hintText` prop and parse it if available.
+	 */
+	@Watch('hintText')
+	private parseHintText() {
+		if (this.hintText) {
+			const hintTextObject = constructHintTextObject(this.hintText);
+			this.internalHintText = hintTextObject;
+		}
+	}
+
+	/**
+	 * Listen for app language settings on the window to update the component language.
+	 */
+	@Listen('setAppLanguage', { target: 'window' })
+	handleSetAppLanguage(event: CustomEvent<Language>) {
+		if (!this.language) {
+			this.language = validateLanguage(event);
+		}
+	}
+
+	/**
+	 * Listen for header language toggling events to update the language dynamically.
+	 */
+	@Listen('headerLanguageToggled', { target: 'window' })
+	handleHeaderLanguageToggled(event: CustomEvent<Language>) {
+		this.language = validateLanguage(event);
+	}
+
+	/**
+	 * Determines the badge color based on the current `taskStatusState`.
+	 */
+	private getBadgeColor(): 'black' | 'light-teal' | 'grey' {
+		return BadgeColors[this.taskStatusState];
+	}
+
+	/**
+	 * Provides a translated task status string based on the current language.
+	 */
+	private getTranslatedTaskStatus(): string {
+		return translations.taskStatus[this.taskStatusState][this.language === 'fr' ? 'fr' : 'en'];
+	}
+
+	/**
+	 * Renders hint text if available.
+	 */
+	private renderHintText() {
+		if (this.internalHintText) {
+			return (
+				<ontario-hint-text
+					hint={this.internalHintText.hint}
+					hintContentType={this.internalHintText.hintContentType}
+					ref={(el) => (this.hintTextRef = el)}
+				></ontario-hint-text>
+			);
+		}
+		return null;
+	}
+
+	/**
+	 * Renders the task label and status content.
+	 */
+	private renderTaskContent() {
+		return (
+			<div class="ontario-task__content">
+				<h2 id="task-label" class="ontario-task__label" aria-label={this.label}>
+					{this.label}
+				</h2>
+				{this.taskStatusState && (
+					<ontario-badge
+						class="ontario-task__badge"
+						role="status"
+						aria-label={`Task status: ${this.getTranslatedTaskStatus()} (${this.taskStatusState})`}
```

Comment:
The text _Task status_ is currently just English and should be handled as a translation.